src/libraries/System.IO.Compression/tests/ZipArchive/zip_LargeFiles.cs

```diff
@@ -44,5 +45,81 @@ public static void UnzipOver4GBZipFile()
+        public static void CheckZIP64VersionIsSet_ForSmallFilesAfterBigFiles()
+        {
+            // issue #94899
+
+            byte[] largeBuffer = GC.AllocateUninitializedArray<byte>(1_000_000_000); // 1 GB
+            byte[] smallBuffer = GC.AllocateUninitializedArray<byte>(1000);
+
+            string zipArchivePath = Path.Combine(Path.GetTempPath(), "over4GB.zip");
+            DirectoryInfo tempDir = Directory.CreateDirectory(Path.Combine(Path.GetTempPath(), "over4GB"));
+
+            try
+            {
+                using FileStream fs = File.Open(zipArchivePath, FileMode.Create, FileAccess.ReadWrite);
+                const string LargeFileName = "largefile";
+                const string SmallFileName = "smallfile";
+                const uint ZipLocalFileHeader_OffsetToVersionFromHeaderStart = 4;
+                const ushort Zip64Version = 45;
+
+                {
+                    // Create
+
+                    using var archive = new ZipArchive(fs, ZipArchiveMode.Create, true);
+                    ZipArchiveEntry file = archive.CreateEntry(LargeFileName, CompressionLevel.NoCompression);
+
+                    using (Stream stream = file.Open())
+                    {
+                        // Write 5GB of data
+
+                        stream.Write(largeBuffer);
+                        stream.Write(largeBuffer);
+                        stream.Write(largeBuffer);
+                        stream.Write(largeBuffer);
+                        stream.Write(largeBuffer);
+                    }
+
+                    file = archive.CreateEntry(SmallFileName, CompressionLevel.NoCompression);
+
+                    using (Stream stream = file.Open())
+                    {
+                        stream.Write(smallBuffer);
+                    }
+                }
+
+                fs.Position = 0;
+
+                {
+                    // Validate
+
+                    using var reader = new BinaryReader(fs);
+                    using var archive = new ZipArchive(fs, ZipArchiveMode.Read);
+                    FieldInfo offsetOfLHField = typeof(ZipArchiveEntry).GetField("_offsetOfLocalHeader", BindingFlags.NonPublic | BindingFlags.Instance);
+
+                    if (offsetOfLHField is null || offsetOfLHField.FieldType != typeof(long))
+                    {
+                        Assert.Fail("Cannot find the private field of _offsetOfLocalHeader in ZipArchiveEntry or the type is not long. Code may be changed after the test is written.");
+                    }
+
+                    foreach (ZipArchiveEntry entry in archive.Entries)
+                    {
+                        fs.Position = (long)offsetOfLHField.GetValue(entry) + ZipLocalFileHeader_OffsetToVersionFromHeaderStart;
+                        ushort versionNeeded = reader.ReadUInt16();
+
+                        Assert.True(versionNeeded >= Zip64Version, "ZIP64 version is not set for files with LH at >4GB offset.");
```

Comment:
Zip64Version is the last value in the enum: https://github.com/dotnet/runtime/blob/f0117c96ace4d475af63bce80d8afa31a740b836/src/libraries/System.IO.Compression/src/System/IO/Compression/ZipVersion.cs#L12

We should not be seeing values greater than that. If we do, this should fail because we either have a bug or we're unexpectedly getting an unsupported value.

So this should be:

```suggestion
                        Assert.Equal(Zip64Version, versionNeeded, "ZIP64 version is not set for files with LH at >4GB offset.");
```

_(I am assuming an optional string message can be passed to `Assert.Equal` as a 3rd argument, please confirm that for me.)_

Reply:
`Assert.Equal` doesn't have a 3 param overload. I've changed the line into:

    Assert.True(versionNeeded == Zip64Version, "Version is not ZIP64 for files with Local Header at >4GB offset.");